htdocs/core/class/extrafields.class.php

```diff
@@ -857,29 +857,32 @@ function showInputField($key, $value, $moreparam='', $keyprefix='', $keysuffix='
 			} else {
 				$checked=' value="1" ';
 			}
-			$out='<input type="checkbox" class="flat '.$showsize.' maxwidthonsmartphone" name="'.$keysuffix.'options_'.$key.$keyprefix.'" '.$checked.' '.($moreparam?$moreparam:'').'>';
+			$out='<input type="checkbox" class="flat '.$showsize.' maxwidthonsmartphone" name="'.$keyprefix.'options_'.$key.$keysuffix.'" '.$checked.' '.($moreparam?$moreparam:'').'>';
 		}
 		elseif ($type == 'price')
 		{
-			$out='<input type="text" class="flat '.$showsize.' maxwidthonsmartphone" name="'.$keysuffix.'options_'.$key.$keyprefix.'" value="'.price2num($value).'" '.($moreparam?$moreparam:'').'> '.$langs->getCurrencySymbol($conf->currency);
+			$out='<input type="text" class="flat '.$showsize.' maxwidthonsmartphone" name="'.$keyprefix.'options_'.$key.$keysuffix.'" value="'.price2num($value).'" '.($moreparam?$moreparam:'').'> '.$langs->getCurrencySymbol($conf->currency);
 		}
 		elseif ($type == 'double')
 		{
-			if (!empty($value)) {
+			// Needed to test if value comes from $_POST
+			// The real form will be "50,00" and price needs to have "50.00"
+			$valPosted = GETPOST($keyprefix.'options_'.$key.$keysuffix);
+			if (!empty($value) && empty($valPosted)) {
 				$value=price($value);
 			}
-			$out='<input type="text" class="flat '.$showsize.' maxwidthonsmartphone" name="'.$keysuffix.'options_'.$key.$keyprefix.'" value="'.$value.'" '.($moreparam?$moreparam:'').'> ';
+			$out='<input type="text" class="flat '.$showsize.' maxwidthonsmartphone" name="'.$keyprefix.'options_'.$key.$keysuffix.'" value="'.$value.'" '.($moreparam?$moreparam:'').'> ';
```

Comment:
If posted value was "50,00", we must keep "50,00" into the form (user must see same data than posted)
If is when we want to use the data that we must convert it using   price2num($datafrompost)

Reply:
The problem comes not from actioncom card page. It comes from all double extrafields. That's why i manage it here. I prefered this code because it was prettier than : 

`// Needed to test if value comes from $_POST for Actioncom and Thirdparty card
			// Because some fields refresh page with $_POST values
			// The real form will send "50,00" and price function needs to have "50.00" (data from database)
			// If currentval was posted, we will apply a price2num function before
			$valPosted = GETPOST($keyprefix.'options_'.$key.$keysuffix);
			if(!empty($valPosted)) {
				$value=price2num($value);
			}
			if (!empty($value)) {
				$value=price($value);
			}`

Do you want me to change for this code ?